binary_search_tree/tree.py

```diff
@@ -14,44 +14,162 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def add_helper(self, current_node, key, value):
+        if current_node == None:
+            return TreeNode(key, value)
+
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(current_node.right, key, value)
+        return current_node
+
+    # Time Complexity: O(n), O(h)
+    # Space Complexity: O(1)
     def add(self, key, value = None):
```

Comment:
👍 However since your helper is recursive the space complexity here is O(log n) or O(h).